src/service/recorder/influxGravityDataRecorder.ts

```diff
@@ -0,0 +1,43 @@
+import {InfluxDB} from 'influx';
+import MQTTTopicClient from '../../mqtt/client';
+import {TransformerFn} from '../thingy';
+import {DataRecorder} from './dataRecorder';
+
+export class InfluxGravityDataRecorder extends DataRecorder {
+    private record = false;
+
+    // simple limit for db writes
+    private frameCount = 0;
+    private nthFrame = 50;
+
+    constructor(mqttClient: MQTTTopicClient, private influxClient: InfluxDB, protected device: any) {
+        super(mqttClient, device);
+    }
+
+    public start(measurement: string, tags: any = {}, transformer: TransformerFn) {
+        this.record = true;
+        this.mqttClient.onTopicMessage(this.topicDefinitions.gravity, (async (message: Buffer) => {
+            // FIXME: THIS is not accessible at callback execution time
+            // this.frameCount++;
+            // if (!this.isRecording() || this.frameCount % this.nthFrame != 0) {
+            //     return;
+            // }
```

Comment:
THIS is accessible and references correctly the recorder instance. When creating the anonymous function THIS is bound.

Reply:
My tests gave opposite results, `this.isRecording()` was always false. It would be best to do a peer debugging session for this ;)

Reply:
Strange. "It works on my machine" ;) Maybe you didn't stop the recorder? `this.isRecording()` is only false after stopping. We can do a debugging session if you want.